src/main/java/com/deavensoft/paketomat/city/CityController.java

```diff
@@ -0,0 +1,126 @@
+
+import com.deavensoft.paketomat.center.model.City;
+import com.deavensoft.paketomat.city.dto.CitiesDto;
+import com.deavensoft.paketomat.city.dto.CityDto;
+import com.deavensoft.paketomat.mapper.CitiesMapper;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.squareup.okhttp.OkHttpClient;
+import com.squareup.okhttp.Request;
+import com.squareup.okhttp.Response;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+
+
+@RestController
+@Slf4j
+@RequestMapping("api/cities")
+public class CityController {
+    private final CityServiceImpl cityServiceImpl;
+    private final CitiesMapper mapper;
+
+    @Autowired
+    public CityController(CityServiceImpl cityServiceImpl, CitiesMapper mapper) {
+        this.cityServiceImpl = cityServiceImpl;
+        this.mapper = mapper;
+    }
+
+    @GetMapping
+    @Operation(summary = "Get cities", description = "Get all cities")
+    @ApiResponse(responseCode = "200", description = "All cities are returned")
+    public List<City> getAllCities() {
+        return cityServiceImpl.getAllCities();
+    }
+
+    @PostMapping
+    @Operation(summary = "Add new city", description = "Add new city to the database")
+    @ApiResponse(responseCode = "200", description = "New city added")
+    public void save(@RequestBody City c) {
+        cityServiceImpl.save(c);
+    }
+
+    @GetMapping(path = "/{id}")
+    @Operation(summary = "Get city", description = "Get city with specified id")
+    @ApiResponse(responseCode = "200", description = "City with specified id returned")
+    public Optional<City> findCityById(@PathVariable(name = "id") long id) {
+        return cityServiceImpl.findCityById(id);
+    }
+
+    @DeleteMapping(path = "/{id}")
+    @Operation(summary = "Delete city", description = "Delete city with specified id")
+    @ApiResponse(responseCode = "200", description = "City with specified id deleted")
+    public int deleteCity(@PathVariable(name = "id") Long id) {
+        Optional<City> c = findCityById(id);
+        if (c.isEmpty())
+           log.info("there is no city with id:" + id);
+        if(!c.isEmpty()) {
+            cityServiceImpl.deleteCity(c.get());
+            log.info("City deleted");
+        }
+        return 1;
+    }
+
+    @GetMapping(path = "/check")
+    public String checkCities() throws IOException {
+        OkHttpClient client = new OkHttpClient();
+
+        Request request = new Request.Builder()
+                .url("https://countries-cities.p.rapidapi.com/location/country/RS/city/list?page=3&per_page=100&population=1501")
+                .get()
+                .addHeader("X-RapidAPI-Key", "c427cbdd50msh072fe84a375af9cp1f2596jsn0f5919dacf12")
```

Comment:
Add in the configuration file (application.yml)

Reply:
Done.